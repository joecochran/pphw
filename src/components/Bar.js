import React, { Component } from 'react';

class Bar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filled: false,
    }
  }

  handleDrop = (event) => {
    this.setState({ filled: true })
    this.props.handleDrop(event);
  }

  render() {
    return (
      <div className={"c-lane__bar" + (this.state.filled ? ' c-lane__bar--filled' : '')} onDrop={this.handleDrop} onDragEnter={(event) => {event.preventDefault()}} onDragOver={(event) => {event.preventDefault()}}>{this.state.filled && (this.props.label)}</div>
    )
  }
  
}

export default Bar;
