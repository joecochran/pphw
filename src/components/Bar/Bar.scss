.c-bar {
  background: #fff;
  height: 38px;
  width: 383px;
  border-radius: 4px;
  border: 1px dashed #C1C4C7;
  position: absolute;
  top: 8px;
  left: 8px;
}
.c-bar:nth-of-type(2) {
  left: 248px;
  top: 54px;
}
.c-bar:nth-of-type(3) {
  left: 418px;
}

.c-bar--filled {
  background-color: #1973BA;
  border: 1px solid #1973BA;
  pointer-events: none;
  display: flex;
  align-items: center;
  padding-left: 16px;
  font-family: Helvetica;
  font-size: 16px;
  color: #FFFFFF;
  line-height: 22px;
}
