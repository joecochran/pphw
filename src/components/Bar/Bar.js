import React, { Component } from 'react';
import './Bar.scss';

class Bar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filled: false,
    }
  }

  handleDrop = (event) => {
    event.preventDefault();
    this.setState({ filled: true })
    this.props.handleDrop(event);
  }

  render() {
    return (
      <div
        className={"c-bar" + (this.state.filled ? ' c-bar--filled' : '')} 
        onDrop={this.handleDrop} 
        onDragEnter={(event) => {event.preventDefault()}} 
        onDragOver={(event) => {event.preventDefault()}}>
          {this.state.filled && (this.props.label)}
        </div>
    )
  }
  
}

export default Bar;
