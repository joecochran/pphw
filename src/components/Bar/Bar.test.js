import React from 'react';
import Bar from './Bar';
import { shallow } from 'enzyme';

describe('<Bar>', () => {
  const wrapper = shallow(<Bar handleDrop={() => {}} />);
  const eventMock = {
    preventDefault: () => {}
  }

  it('matches snapshot upon init', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('matches snapshot after drop', () => {
    wrapper.simulate('dragenter', eventMock);
    wrapper.simulate('dragover', eventMock);
    wrapper.simulate('drop', eventMock);
    expect(wrapper).toMatchSnapshot();
  })
})
