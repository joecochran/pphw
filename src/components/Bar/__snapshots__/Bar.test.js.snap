// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar> matches snapshot after drop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bar
    handleDrop={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "c-bar c-bar--filled",
      "onDragEnter": [Function],
      "onDragOver": [Function],
      "onDrop": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "c-bar c-bar--filled",
        "onDragEnter": [Function],
        "onDragOver": [Function],
        "onDrop": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Bar> matches snapshot upon init 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bar
    handleDrop={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
      "className": "c-bar",
      "onDragEnter": [Function],
      "onDragOver": [Function],
      "onDrop": [Function],
    },
    "ref": null,
    "rendered": false,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
        "className": "c-bar",
        "onDragEnter": [Function],
        "onDragOver": [Function],
        "onDrop": [Function],
      },
      "ref": null,
      "rendered": false,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
